drivers/include/sdmmc/sdmmc.h

```diff
@@ -0,0 +1,1905 @@
+    uint16_t SD_CARD_TYPE:16;           /**< [495:480] Section 4.10.2, Table 4-44 */
+    uint16_t ERASE_SIZE:16;             /**< [423:408] Section 4.10.2.5, Table 4-49 */
+    uint8_t  SPEED_CLASS:8;             /**< [447:440] Section 4.10.2.2, Table 4-45 */
+    uint8_t  PERFORMANCE_MOVE:8;        /**< [439:432] Section 4.10.2.3, Table 4-46 */
+    uint8_t  VIDEO_SPEED_CLASS:8;       /**< [391:384] Section 4.10.2.10, Table 4-54 */
+    uint8_t  ERASE_TIMEOUT:6;           /**< [407:402] Section 4.10.2.6, Table 4-50 */
+    uint8_t  ERASE_OFFSET:2;            /**< [401:400] Section 4.10.2.7, Table 4-51 */
+    uint8_t  UHS_SPEED_GRADE:4;         /**< [399:396] Section 4.10.2.8, Table 4-52 */
+    uint8_t  UHS_AU_SIZE:4;             /**< [395:392] Section 4.10.2.9, Table 4-3 */
+    uint8_t  AU_SIZE:4;                 /**< [431:428] Section 4.10.2.4, Table 4-47 */
+    uint8_t  DAT_BUS_WIDTH:2;           /**< [511:510] Section 4.10.2, Table 4-44 */
+    uint8_t  SECURED_MODE:1;            /**< [509] Section 4.10.2, Table 4-44 */
+} sdmmc_sd_status_t;
+
+/** @} */
+
+/**
+ * @name CID register (SD Memory and MMC Cards)
+ * @{
+ */
+
+/** CID register size in bytes (CID is 128 bit) */
+#define SDMMC_CID_REG_SIZE      (16)
+
+#define SDMMC_CID_OID_SIZE_SD   (2)   /**< OID (OEM/Application ID) size in byte (SD) */
+#define SDMMC_CID_PNM_SIZE_SD   (5)   /**< PNM (Product name) size in byte (SD) */
+#define SDMMC_CID_PNM_SIZE_MMC  (6)   /**< PNM (Product name) size in byte (MMC) */
+
+/**
+ * @brief   CID register structure (SD Memory Cards)
+ *
+ * @see Physical Layer Simplified Specification Version 9.00, Section 5.2,
+ *      Table 5.2 [[sdcard.org](https://www.sdcard.org)]
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t MID;                      /**< Manufacturer ID */
+    char OID[SDMMC_CID_OID_SIZE_SD];  /**< OEM/Application ID */
+    char PNM[SDMMC_CID_PNM_SIZE_SD];  /**< Product name */
+    uint8_t PRV;                      /**< Product revision */
+    uint32_t PSN;                     /**< Product serial number in big-endian order */
+    uint16_t MDT;                     /**< Manufacturing date in big-endian order */
+    uint8_t CID_CRC;                  /**< CRC7 checksum including bit 0 */
+} sdmmc_cid_sd_t;
+
+/**
+ * @brief   CID register structure (MMC)
+ *
+ * @see JEDEC Standard No. JESD84-B42, MultiMediaCard (MMC) Electrical
+ *      Standard, High Capacity (MMCA, 4.2), Section 8.2, Table 30
+ *      [[jedec.org](https://www.jedec.org)]
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t MID;                      /**< Manufacturer ID */
+    uint16_t OID;                     /**< OEM/Application ID in big-endian order */
+    char PNM[SDMMC_CID_PNM_SIZE_MMC]; /**< Product name */
+    uint8_t PRV;                      /**< Product revision */
+    uint32_t PSN;                     /**< Product serial number in big-endian order */
```

Comment:
Would it makes sense to use type like `be_uint32_t` for big endian representation ?
This could also apply to other variables in this file.

Reply:
Yeah, it might be a `be_uint32_t`. But the only advantage is that it documents by type that it is in BE order. Since there is no implicit conversion if needed, you have to use explicitly `byteorder_ntohl` which makes no difference to `nothl`. If you think it is better to use `be_uint32_t` and `be_uint16_t` here, I'm fine with it.

Reply:
I don't see any other structure where it could makes sense. Either the data structures are bit fields that are directly mapped to a single the 32 bit word returned in one of the R[n] response registers which is already in host byte order (e.g. `sdmmc_card_status_t` or the data structures can't be mapped directly to read buffer in big endian order because a couple of members are neither byte nor word-aligned (e.g. `sdmmc_csd_v1_t`). With the latter data structures, the members must be filled with byte-wise bit operations anyway.

Reply:
IMHO, it is good practice so please go ahead. It also help when you tried to modify a driver after a few months or so and you forget about it :) I had some issues like this with SCSI implementation for USB MSC because of this and my golden fish memory.